util/fusesoc_build.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""Command-line tool to add the calling interpreter to fusesoc's PATH, so it
+is used for generators.
+"""
+
+
+import os
+import sys
+from fusesoc.main import main
+
+if __name__ == "__main__":
+    # First, ensure the calling interpreter is on the PATH first, so any
+    # generators asking /usr/bin/env for python3 will use the same version.
+    path_env = os.environ["PATH"]
+    if path_env is not None:
+        path_env = ":" + path_env
+    path_env = os.path.dirname(sys.executable) + path_env
+    os.environ["PATH"] = path_env
```

Comment:
Is there no way to set the PATH up from Bazel?

Reply:
There certainly are ways, but it would likely involve recreating a bunch of work that rules_python already does and hides under the covers.

In my first attempts here, I was able to extract the PyRuntimeInfo provider in Starlark, but it yielded a relative path to the interpreter. It felt unreliable to add a relative path to the PATH variable, so I stopped there, before even trying to bring in all the other dependencies. Making use of py_binary and the existing dependency framework seemed more straightforward.

Reply:
Sounds sensible. This solution is nice and simple.